franka-interface/src/skills/joint_torque_skill.cpp

```diff
@@ -0,0 +1,141 @@
+    std::array<double, 16> pose_desired; 
+
+    RunLoopSharedMemoryHandler* shared_memory_handler = run_loop->get_shared_memory_handler();
+    RunLoopProcessInfo* run_loop_info = shared_memory_handler->getRunLoopProcessInfo();
+    boost::interprocess::scoped_lock<boost::interprocess::interprocess_mutex> lock(
+                                    *(shared_memory_handler->getRunLoopProcessInfoMutex()),
+                                    boost::interprocess::defer_lock);
+    SensorDataManager* sensor_data_manager = run_loop->get_sensor_data_manager();
+
+    std::cout << "Will run the control loop\n";
+    //Unneccessary for torque skill
+    //JointTrajectoryGenerator* joint_trajectory_generator = dynamic_cast<JointTrajectoryGenerator*>(traj_generator_);
+    //SetInternalImpedanceFeedbackController* internal_feedback_controller = dynamic_cast<SetInternalImpedanceFeedbackController*>(feedback_controller_);
+
+    //if(joint_trajectory_generator == nullptr) {
+    //    throw std::bad_cast();
+    //}
+
+    std::function<franka::Torques(const franka::RobotState&, franka::Duration)>
+        joint_torque_callback = [&](
+        const franka::RobotState& robot_state,
+        franka::Duration period) -> franka::Torques {
+	    current_period_ = period.toSec();
+            time += current_period_;        
+            if(time == 0.0 ){
+                //////
+                //traj_generator_->initialize_trajectory(robot_state, SkillType::JointTorqueSkill);
+                for(int i = 0; i < 7; i++) {
+                    current_joint_torques_[i] = 0;
+                    previous_joint_torques_[i] = 0;
+                }
+
+                try {
+                    if (lock.try_lock()) {
+                    run_loop_info->set_time_skill_started_in_robot_time(robot_state.time.toSec());
+                    lock.unlock();
+                    } 
+                } catch (boost::interprocess::lock_exception) {
+                    // Do nothing
+                    }                
+            }
+
+            if (robot_state_data->mutex_.try_lock()) {
+            robot_state_data->counter_ += 1;
+            robot_state_data->time_ =  period.toSec();
+            robot_state_data->has_data_ = true;
+            robot_state_data->mutex_.unlock();
+            }
+
+            //traj_generator_->time_ = time;//////
+            //traj_generator_->dt_ = current_period_;//////
+
+            log_counter += 1;            
+            try {
+                sensor_data_manager->getSensorBufferGroupMutex()->try_lock();
+                traj_generator_->parse_sensor_data(robot_state);
+                termination_handler_->parse_sensor_data(robot_state);
+                sensor_data_manager->getSensorBufferGroupMutex()->unlock();
+                } catch (boost::interprocess::lock_exception) {
+                    }
+            log_counter += 1;
+            if (log_counter % 1 == 0) {
+            pose_desired = robot_state.O_T_EE_d;
```

Comment:
why is this logging required here? also pose desired may not really be correct at this stage.

Reply:
I thought this was necessary for logging robot states. I double checked O_T_EE_d and O_T_EE are the same in this stage